### PR TITLE
SerializationService toObject for InstanceOfPredicate causes NullPointerException

### PR DESCRIPTION
**Pull request by ihsandemir**
When using InstanceOfPredicate, the server side toObject method causes NullPointerException, since the classloader is not set and it is null. This fix uses the default classloader if it is not set by the config to avoid the null pointer exception.

### CODE REVIEW

**Comment by sancar** [comment]
@ihsandemir Can you test the same thing also for node ? I took a quick look at the code, and it looks like same test will fail for server too.

File: hazelcast-client/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java [hunk 1]
```diff
@@ -64,6 +64,9 @@ public InternalSerializationService createSerializationService(byte version) {
         try {
             ClientConfig config = client.getClientConfig();
             ClassLoader configClassLoader = config.getClassLoader();
+            if (null == configClassLoader) {
```
Comment (sancar) [inline hazelcast-client/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java]: Using our classLoaderUtil for this seems like correct solution
`klass = ClassLoaderUtil.loadClass(in.getClassLoader(), klassName);`

Comment (ihsandemir) [inline hazelcast-client/src/main/java/com/hazelcast/client/impl/DefaultClientExtension.java]: @sancar Ok, changed as suggested. Please review again.

**Comment by ihsandemir** [comment]
verify

**Comment by ihsandemir** [comment]
@sancar Added the test.

**Comment by sancar** [comment]
👍

**Pull request merged**